Docx Writer/Reference: Add keepNext to objects w/ captions

lierdakil opened the pull request:
This changes `TableCaption` style, adding `w:keepNext` option, and adds `Figure` style to figure images and `FigureWithCaption` to figures with nonempty captions, which is based on `Figure`, and additionally has `w:keepNext` set as well

With this, caption and table/figure are not separated at page boundaries, as per #2031.

lierdakil commented:
This feels a bit clunky with two separate styles for figures. If anyone has a better idea, I'm open to suggestions.

One option might be to hard-add `w:keepNext` to image paragraph itself, not depending on style, but I feel that would limit customisation options.

mpickering commented:
I think the section in the documentation needs to be updated as well.
On 29 Mar 2015 06:11, "Nikolay Yakimov" <notifications@github.com> wrote:

> This feels a bit clunky with two separate styles for figures. If anyone
> has a better idea, I'm open to suggestions.
>
> One option might be to hard-add w:keepNext to image paragraph itself, not
> depending on style, but I feel that would limit customisation options.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jgm/pandoc/pull/2035#issuecomment-87359371>.
>

lierdakil commented:
Ok, updated README with new docx style names. I've also updated those that were changed in 1cb601d2885df09fe07533006c06d8a603c3020d.
Docx style semantics probably should be documented somewhere on wiki as well, as it's not immediately intuitive.

tolot27 commented:
Thanks for implementing this. I don't see any other option to implement this as long as Word has no `w:keepPrevious` (which is suggested to Microsoft for years, btw).
Word itself hardcodes the `w:keepNext` if you add a caption to a figure (see context menue of figure). I would follow the same procedure and do not "over style". Having a separate figure style is one step further already.
